.github/workflows/publish-major-minor.yml

```
@@ -0,0 +1,51 @@
+name: "Publish Major-Minor-Tags"
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  push-tags:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install packaging
+        run: python -m pip install -U packaging --user
+      - name: Get versions
+        id: get_versions
+        shell: python
+        run: |
+          from packaging.version import parse
+
+          tag_ref = "${{ github.ref }}"
+          tag_name = tag_ref.split("/")[-1]
+          version = parse(tag_name)
+          print(f"tag_name: {tag_name}")
+          print(f"version: {version}")
+          if not version.is_prerelease:
+              print("Creating new major and minor tags!")
+              print(f"::set-output name=original_tag_name::{tag_name}")
+              print(f"::set-output name=major_version::v{version.major}")
+              print(f"::set-output name=minor_version::v{version.major}.{version.minor}")
+          else:
+              print("No tags created (dev or pre version)!")
+
+      - name: Push Tags Version
+        if: steps.get_versions.outputs.original_tag_name != ''
+        env:
+          original_tag_name: ${{ steps.get_versions.outputs.original_tag_name }}
+          major_version: ${{ steps.get_versions.outputs.major_version }}
+          minor_version: ${{ steps.get_versions.outputs.minor_version }}
+        run: |
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+          git config user.name 'github-actions[bot]'
+          git tag --annotate '${{ env.major_version }}' \
+            --message='original tag: ${{ env.original_tag_name }}'
```

have you considered appending the message from the original tag too?
```suggestion
            --message='original tag: ${{ env.original_tag_name }}' \
            --message="$(git tag -l --format='%(contents)' '${{ env.original_tag_name }}')"
```

---

This would only work as intended if the tag had an additional message to it or?
E.g.  `v1.4.0` works fine
```console
$ git tag -l --format='%(contents)' 'v1.4.0'
Merge PR #39

This change exposes an extra GitHub Action input called `verbose`. It
allows to see `twine upload` detailed output if needed for debugging.
```

while `v1.4.1` doesn't

```console
$ git tag -l --format='%(contents)' 'v1.4.1'
v1.4.1
-----BEGIN PGP SIGNATURE-----
.
.
.
-----END PGP SIGNATURE-----
```

IMHO adding a link to the release would be more informative.

---

In the second case, `v1.4.1` is also an additional message because all signed tags are also annotated. While I understand the sentiment of having a link, I'm not sure if it could be achieved easily. Another thing that would be useful is the commit hash where the tag points at (security-wise).

---

With some conditions, creating a link would be pretty straight forward:
- release name always needs to be the same as tag name
- workflow triggers on 
   ```yaml
   on:
     release:
      types: [created, edited]
   ```

If those conditions are met, the link would be 
`https://github.com/pypa/gh-action-pypi-publish/releases/tag/${{ env.original_tag_name }}`

As for adding the SHA1 of the commit, this only provides additional information in a CLI context, since the github website already points to the underlying commit ([see test tag v2.0.0](https://github.com/s-weigand/gh-action-pypi-publish/releases)).

---

> As for adding the SHA1 of the commit, this only provides additional information in a CLI context, since the github website already points to the underlying commit ([see test tag v2.0.0](https://github.com/s-weigand/gh-action-pypi-publish/releases)).

Besides being a pointer, it's also proof that it was created by the workflow and not manually, overriding it to some malicious commit.